apiserver/facades/client/sshclient/shim.go

```diff
@@ -36,10 +38,19 @@ type NetworkService interface {
 	GetAllSubnets(ctx context.Context) (network.SubnetInfos, error)
 }
 
+// ApplicationService is the interface that is used to interact with the
+// applications.
+type ApplicationService interface {
+	// GetUnitMachineName gets the name of the unit's machine.
+	//
+	// The following errors may be returned:
+	//   - [applicationerrors.UnitNotFound] if the unit cannot be found.
+	GetUnitMachineName(context.Context, unit.Name) (machine.Name, error)
```

Comment:
The returned errors here don't match the ones in the service.
```suggestion
	// GetUnitMachineName gets the name of the unit's machine.
	//
        // The following errors may be returned:
        //   - [applicationerrors.UnitMachineNotAssigned] if the unit does not have a
        //     machine assigned.
        //   - [applicationerrors.UnitNotFound] if the unit cannot be found.
        //   - [applicationerrors.UnitIsDead] if the unit is dead.
	GetUnitMachineName(context.Context, unit.Name) (machine.Name, error)
```